qgreenland/util/luigi/tasks/main.py

```diff
@@ -161,9 +162,14 @@ def run(self):
         with temporary_path_dir(self.output()) as temp_path:
             shutil.copy2(input_fp, temp_path / final_fn)
 
-            # Create layer provenance file. This is not an "AncillaryFile" job
-            # because we need one file per layer.
-            with open(temp_path / 'provenance.txt', 'w') as provenance_file:
-                provenance_file.write(
-                    steps_to_provenance_text(self.layer_cfg.steps),
-                )
+            # Create layer provenance and metadata files. These are not
+            # "AncillaryFile" jobs because we need one file per layer.
+            write_provenance_file(
+                layer_cfg=self.layer_cfg,
+                filepath=temp_path / 'provenance.txt',
+            )
+
+            write_metadata_file(
+                layer_cfg=self.layer_cfg,
+                filepath=temp_path / 'metadata.txt',
+            )
```

Comment:
:thinking: Provenance is also metadata...Should these be one file? A different name for `metadata.txt`?

Reply:
True. I also have thoughts about whether metadata should be text. What about JSON? Provenance being in its own file makes it more machine-readable than if we put it in its own section of metadata.txt; with JSON we could extract the provenance programmatically. :shrug:

Reply:
True, although I think since this is primarily for users who are viewing data on their file system, txt makes more sense than json. This is meant to be human-readable as opposed to machine readable.

Reply:
The `layer_list.csv` also provides a machine-readable way to extract this sort of information if a user wishes to do so.

Reply:
(Going to leave as `metadata.txt` for now.